teuthology/suite/run.py

```diff
@@ -583,12 +587,12 @@ def schedule_suite(self):
                                seed=self.args.seed)
         generated = len(configs)
         log.info(f'Suite {suite_name} in {suite_path} generated {generated} jobs (not yet filtered or merged)')
-        configs = config_merge(configs,
+        configs = list(config_merge(configs,
```

Comment:
>    The var being left as a generator led to all jobs being filtered out of    some suites, rendering them broken.

! Can you show me an example?

Reply:
Try this:
`teuthology-suite -v --teuthology-branch main --ceph-repo https://github.com/ceph/ceph.git --suite-repo https://github.com/ceph/ceph.git -c main -m testnode -n 100 --suite teuthology:no-ceph --suite-branch main --dry-run`

Reply:
:open_mouth: is there some esoteric python gotcha I tripped on or is this just as baffling to you?

Reply:
I didn't step through everything as thoroughly as I could have so I don't know exactly how the generator is being exhausted. But, generators in python can only be iterated over once:
```
>>> l = [n for n in range(5)]  # just a regular list
>>> list(l)
[0, 1, 2, 3, 4]
>>> list(l)
[0, 1, 2, 3, 4]
>>> g = (n for n in range(5))  # this one is a generator
>>> g
<generator object <genexpr> at 0x10464f7b0>
>>> list(g)
[0, 1, 2, 3, 4]
>>> list(g)
[]  # empty!
```

Reply:
Yes I'm aware of that behavior. I don't have an explanation other than a bug in Python. The `configs` generator is only iterated once AFAICT.

Reply:
Oh, I see it's because of `-n`. There is a while loop around `collect_jobs`.

Guess we can't use a generator!